src/modules/tests/tests.service.ts

```
@@ -463,4 +469,142 @@ export class TestsService {
+    );
+
+    // Calculate final result based on grading method
+    const submittedAttempts = allAttempts.filter(a => a.status === AttemptStatus.SUBMITTED);
+    let finalScore: number = 0;
+    let finalResult: string | null = null;
+    let finalAttemptId: string = null;
+
+    if (submittedAttempts.length > 0) {
+      switch (test.attemptsGrading) {
+        case AttemptsGradeMethod.FIRST_ATTEMPT:
+          const firstAttempt = submittedAttempts[0]; // First by start time
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)